lib/operations/db_ops.js

```diff
@@ -203,12 +206,12 @@ function createIndex(db, name, fieldOrSpec, options, callback) {
      * 197 = 'InvalidIndexSpecificationOption' (`_id` with `background: true`)
      */
     if (
-      err.code === 67 ||
-      err.code === 11000 ||
-      err.code === 85 ||
-      err.code === 86 ||
-      err.code === 11600 ||
-      err.code === 197
+      /** @type {Error & {code?:number}} */ (err).code === 67 ||
+      /** @type {Error & {code?:number}} */ (err).code === 11000 ||
+      /** @type {Error & {code?:number}} */ (err).code === 85 ||
+      /** @type {Error & {code?:number}} */ (err).code === 86 ||
+      /** @type {Error & {code?:number}} */ (err).code === 11600 ||
+      /** @type {Error & {code?:number}} */ (err).code === 197
```

Comment:
I'm guessing this pattern is necessary to fix a specific error; it seems pretty awkward. Is there anything else we could do here, possibly an ignore?

Reply:
I could do `err['code']` or `//@ts-ignore` on each line .code doesn't exist on the type that err is because I believe err here is either a MongoError, Error, or a (error) result object from the db and its only the last one that has `.code` open to whatever we think is best

Reply:
How about
```
    const code = /** @type {Error & {code?:number}} */ (err).code;
    if (
      code === 67 ||
      code === 11000 ||
      code === 85 ||
      code === 86 ||
      code === 11600 ||
      code === 197
    ) {
      return handleCallback(callback, err, result);
    }
```

Reply:
Much cleaner, went with that 😃